crypto/types/src/vrf.rs

```diff
@@ -0,0 +1,220 @@
+use crate::errors::CryptoError::CalculationError;
+use crate::keypairs::{ChainKeypair, Keypair};
+use crate::types::CurvePoint;
+use crate::utils::k256_scalar_from_bytes;
+use hopr_crypto_random::random_bytes;
+use k256::elliptic_curve::hash2curve::{ExpandMsgXmd, GroupDigest};
+use k256::elliptic_curve::sec1::ToEncodedPoint;
+use k256::elliptic_curve::ProjectivePoint;
+use k256::{Scalar, Secp256k1};
+use serde::{Deserialize, Serialize};
+use utils_types::errors::GeneralError::ParseError;
+use utils_types::primitives::Address;
+use utils_types::traits::BinarySerializable;
+
+/// Bundles values given to the smart contract to prove that a ticket is a win.
+///
+/// The VRF is thereby needed because it generates on-demand deterministic
+/// entropy that can only be derived by the ticket redeemer.
+#[derive(Copy, Clone, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct VrfParameters {
+    /// the pseudo-random point
+    pub v: CurvePoint,
+    pub h: Scalar,
+    pub s: Scalar,
+    /// helper value for smart contract
+    pub h_v: CurvePoint,
+    /// helper value for smart contract
+    pub s_b: CurvePoint,
```

Comment:
Could these be in a different type composed of raw vrf params?

Reply:
What do you mean? These two values are secp256k1 elliptic curve points